### PR TITLE
Table cleanup before sub-rows PR

### PR DESCRIPTION
**Pull request by bentleyvk**
Moved hooks and action handlers to separate files as there will be more code inside them after sub-rows PR.
No code inside has changed.

## Checklist

- [ ] Add meaningful unit tests for your component (verify that all lines are covered)
- [x] Verify that all existing tests pass
- [ ] Add component features demo in Storybook (different stories)
- [ ] Approve test images for new stories
- [ ] Add an entry to the changelog for any new components and changes
- [ ] Add screenshots of the key elements of the component
- [ ] Add any additional comments describing the features you've implemented

### CODE REVIEW

File: src/core/Table/Table.tsx [hunk 1]
```diff
@@ -231,202 +231,45 @@ export const Table = <
...
-          ...newState,
-          selectedRowIds: { [action.id]: true } as Record<string, boolean>,
-        };
-      }
-      case TableActions.toggleRowSelected:
-      case TableActions.toggleAllRowsSelected:
-      case TableActions.toggleAllPageRowsSelected: {
-        onSelectHandler(newState, instance);
-        break;
+  const tableStateReducer = React.useCallback(
```
Comment (bentleyvk) [inline src/core/Table/Table.tsx:234]: I am not sure whether to move it to separate file or not. The only down side having it in separate file is that function will have a lot of props.

Comment (mayank99) [reply src/core/Table/Table.tsx:234]: I'd say leave it here.

File: src/core/Table/hooks/index.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+export { EXPANDER_CELL_ID, useExpanderCell } from './useExpanderCell';
+export { SELECTION_CELL_ID, useSelectionCell } from './useSelectionCell';
```
Comment (mayank99) [inline src/core/Table/hooks/index.ts:6]: Do we need to export those ids?

Not sure if you plan to use them in your sub-rows PR or if it's just for futureproofing

**Review by mayank99 — APPROVED** [review]

File: src/core/Table/hooks/index.ts (hunk 2 above)
Comment (bentleyvk) [reply src/core/Table/hooks/index.ts:6]: Yeah, I plan to use them. But if turns out to be a bad idea, I will remove them.

**Pull request merged**